### PR TITLE
Sol5 -> Sol6

### PR DESCRIPTION
**Pull request by Anyhowclick**
- Not sure how to fix travis, need someone's help. Probably Duc
- ./coverage.sh doesn't work at the moment. Some functions fail to be overridden properly by coverage I think...
- ./test/sol4/kyberProxyV1.js tests are broken... need someone to help fix as well

### CODE REVIEW

File: .solcover.js [hunk 1]
```diff
@@ -2,23 +2,23 @@ const fs = require('fs');
 const path = require('path');
 const mv = require('mv');
 
-const v4ArtifactsPath = path.join(__dirname, '.coverageV4Artifacts');
+const oldArtifactsPath = path.join(__dirname, '.coverageArtifacts');
```
Comment (ilanDoron) [inline .solcover.js:5]: I don't follow what does old mean here?
can it be named previous?
or is it old?

File: buidlerConfigSol4.js [hunk 2]
```diff
@@ -1,11 +1,10 @@
 module.exports = {
-  defaultNetwork: "buidlerevm",
```
Comment (ilanDoron) [inline buidlerConfigSol4.js:2]: this changes?

File: contracts/sol6/Dao/IEpochUtils.sol [hunk 3]
```diff
@@ -0,0 +1,7 @@
+pragma solidity 0.6.6;
+
+interface IEpochUtils {
+    function epochPeriodInSeconds() external view returns (uint256);
+
+    function firstEpochStartTimestamp() external view returns (uint256);
+}
```
Comment (ilanDoron) [inline contracts/sol6/Dao/IEpochUtils.sol:7]: last line.

File: contracts/sol6/Dao/KyberFeeHandler.sol [hunk 4]
```diff
@@ -172,10 +171,12 @@ contract KyberFeeHandler is IKyberFeeHandler, Utils4, BurnConfigPermission {
         _;
     }
 
-    function() external payable {
+    receive() external payable {
         emit EthReceived(msg.value);
     }
 
+    fallback() external {}
```
Comment (ilanDoron) [inline contracts/sol6/Dao/KyberFeeHandler.sol:178]: do we need this one?

File: contracts/sol6/IKyberDAO.sol [hunk 5]
```diff
@@ -1,7 +1,9 @@
-pragma solidity 0.5.11;
+pragma solidity 0.6.6;
 
+import "./Dao/IEpochUtils.sol";
 
-interface IKyberDAO {
+
+interface IKyberDAO is IEpochUtils {
```
Comment (ilanDoron) [inline contracts/sol6/IKyberDAO.sol:6]: so now can inherit interface.
its new to me.

File: contracts/sol6/KyberHintHandler.sol [hunk 6]
```diff
@@ -280,7 +300,7 @@ contract KyberHintHandler is IKyberHint, Utils4 {
         if (error == HintErrors.SplitsNotEmptyError) revert("splits must be empty");
     }
 
-    function convertReserveIdToAddress(bytes32 reserveId) internal view returns (address);
+    function convertReserveIdToAddress(bytes32 reserveId) internal view virtual returns (address);
```
Comment (ilanDoron) [inline contracts/sol6/KyberHintHandler.sol:303]: we still need this part?
we convert to address?

File: contracts/sol6/KyberMatchingEngine.sol [hunk 7]
```diff
@@ -31,14 +31,19 @@ contract KyberMatchingEngine is KyberHintHandler, IKyberMatchingEngine, Withdraw
...
         emit KyberStorageUpdated(_kyberStorage);
         kyberStorage = _kyberStorage;
         return true;
     }
 
-    function setNegligbleRateDiffBps(uint256 _negligibleRateDiffBps) external returns (bool) {
+    function setNegligbleRateDiffBps(uint256 _negligibleRateDiffBps)
+        external
+        virtual
+        override
```
Comment (ilanDoron) [inline contracts/sol6/KyberMatchingEngine.sol:44]: what does this override?

File: contracts/sol6/KyberMatchingEngine.sol [hunk 8]
```diff
@@ -69,6 +73,7 @@ contract KyberMatchingEngine is KyberHintHandler, IKyberMatchingEngine, Withdraw
     )
         external
         view
+        override
```
Comment (ilanDoron) [inline contracts/sol6/KyberMatchingEngine.sol:76]: what does it override?

File: buidlerConfigSol4.js (hunk 2 above)
Comment (Anyhowclick) [reply buidlerConfigSol4.js:2]: Not needed, since we only use this file for compilation.

Comment (Anyhowclick) [reply buidlerConfigSol4.js:2]: Same for buidlerConfigSol5.

File: contracts/sol6/IKyberDAO.sol (hunk 5 above)
Comment (Anyhowclick) [reply contracts/sol6/IKyberDAO.sol:6]: Yeah, didn't know too

File: contracts/sol6/Dao/KyberFeeHandler.sol (hunk 4 above)
Comment (Anyhowclick) [reply contracts/sol6/Dao/KyberFeeHandler.sol:178]: I'm not sure when we need fallback() if we have receive()... Not sure what the best practice is here.

File: contracts/sol6/KyberMatchingEngine.sol (hunk 7 above)
Comment (Anyhowclick) [reply contracts/sol6/KyberMatchingEngine.sol:44]: Interface

File: contracts/sol6/KyberHintHandler.sol (hunk 6 above)
Comment (Anyhowclick) [reply contracts/sol6/KyberHintHandler.sol:303]: Hmmmm, since we shift to storage, maybe we can read from there instead?

Comment (Anyhowclick) [reply contracts/sol6/KyberHintHandler.sol:303]: We can discuss this. Cos network side also stores ids to address... if we just get from storage, then we can remove it too.

File: contracts/sol6/Dao/KyberFeeHandler.sol (hunk 4 above)
Comment (Anyhowclick) [reply contracts/sol6/Dao/KyberFeeHandler.sol:178]: "Prefer using a receive function only, whenever possible."
https://solidity.readthedocs.io/en/v0.6.6/060-breaking-changes.html#how-to-update-your-code

Will remove

**Pull request merged**